pyccel/parser/base.py

```diff
@@ -410,40 +410,40 @@ def print_namespace(self):
             print ('{var} \t :: \t {dtype}'.format(var=k, dtype=type(v)))
         print ('-------------------------')
 
-    def view_namespace(self, entry):
-
-        # TODO improve
-
-        try:
-            from tabulate import tabulate
-
-            table = []
-            for (k, v) in self.namespace[entry].items():
-                k_str = '{}'.format(k)
-                if entry == 'imports':
-                    if v is None:
-                        v_str = '*'
-                    else:
-                        v_str = '{}'.format(v)
-                elif entry == 'variables':
-                    v_str = '{}'.format(type(v))
-                else:
-                    raise NotImplementedError('TODO')
-
-                line = [k_str, v_str]
-                table.append(line)
-            headers = ['module', 'target']
-
-#            txt = tabulate(table, headers, tablefmt="rst")
-
-            txt = tabulate(table, tablefmt='rst')
-            print (txt)
-        except NotImplementedError:
-
-            print ('------- namespace.{} -------'.format(entry))
-            for (k, v) in self.namespace[entry].items():
-                print ('{var} \t :: \t {value}'.format(var=k, value=v))
-            print ('-------------------------')
+#    def view_namespace(self, entry):
```

Comment:
Should this function be moved to the obsolete folder rather than commenting it?
That is where we have the other functions which are not tested but are still useful. If it is not useful then maybe it should be removed rather than commented

Reply:
This is a class method, so moving it to the "obsolete" folder does not seem useful unless we also copy the class there. As for removing it, we should ask the original author (@ratnania I believe) if he thinks this function could still be useful.

Reply:
Although it is a class method it doesn't seem to use any private properties of the class. If we passed a namespace (self.namespace) instead of passing the class instance then it could easily be separated from the class

Reply:
OK I will move this function to a separate file in the `obsolete` folder.

Reply:
Actually I don't see how this function was supposed to work. `self` is a `BasicParser` object, hence`self.namespace` is an instance of `Scope`, which has no `__getitem__` method. So the expression `self.namespace[entry]` has no meaning.

@saidctb Maybe `self.namespace` was a dictionary at some point?

Reply:
I suspect it was an OrderedDict at some point. I would guess that the expected options for entry are:
- variables
- classes
- functions
- macros
- headers

And possibly
- imports